lib/internal/bootstrap/switches/is_main_thread.js

```diff
@@ -148,7 +148,7 @@ function getStdin() {
   switch (guessHandleType(fd)) {
     case 'TTY':
       const tty = require('tty');
-      stdin = new tty.ReadStream(fd);
+      stdin = new tty.ReadStream(fd, { highWaterMark: 0 });
```

Comment:
If this means that Node.js starts reading data in situations where it currently does not, that’s a major breaking change – for example, a subprocess that shares the same stdin (e.g. stdio: 'inherit') might not end up reading data because Node.js reads data before it gets the chance.

Reply:
@addaleax yes I agree it’s better without however, https://github.com/nodejs/node/pull/39253#issuecomment-873478298 writes that the other way around is a major breakage.